kernel/src/fs/utils/flock.rs

```diff
@@ -96,49 +96,65 @@ impl Debug for FlockItem {
 /// Represents a list of non-POSIX file advisory locks (FLOCK).
 /// The list is used to manage file locks and resolve conflicts between them.
 pub struct FlockList {
-    inner: Mutex<VecDeque<FlockItem>>,
+    inner: RwMutex<VecDeque<FlockItem>>,
 }
 
 impl FlockList {
     /// Creates a new FlockList.
     pub fn new() -> Self {
         Self {
-            inner: Mutex::new(VecDeque::new()),
+            inner: RwMutex::new(VecDeque::new()),
         }
     }
 
     /// Attempts to set a lock on the file.
+    ///
+    /// If no conflicting locks exist, the lock is set and the function returns Ok(()).
+    /// If a conflicting lock exists:
+    /// - If waker is not None, it is added to the conflicting lock's waitqueue, and the function returns EAGAIN.
+    /// - If waker is None, the function returns EAGAIN.
```

Comment:
```suggestion
    /// If no conflicting locks exist, the lock is set and the function returns `Ok(())`.
    /// If a conflicting lock exists:
    /// - If waker is not `None`, it is added to the conflicting lock's waitqueue, and the function returns `EAGAIN`.
    /// - If waker is `None`, the function returns `EAGAIN`.
```

The backticks are missing.

(I understand that many comments in these two files are poorly written. So to be "consistent" with existing bad comments, you can leave this for future work. But it is worth mentioning).